src/promptflow/promptflow/_utils/flow_utils.py

```diff
@@ -42,3 +44,29 @@ def get_flow_lineage_id(flow_dir: Union[str, PathLike]):
     # hash the value to avoid it gets too long, and it's not user visible.
     lineage_id = hashlib.sha256(lineage_id.encode()).hexdigest()
     return lineage_id
+
+
+def apply_default_value_for_input(inputs: Dict[str, FlowInputDefinition], line_inputs: Mapping) -> Dict[str, Any]:
+    updated_inputs = dict(line_inputs or {})
+    for key, value in inputs.items():
+        if key not in updated_inputs and (value and value.default):
+            updated_inputs[key] = value.default
+    return updated_inputs
+
+
+def handle_line_failures(run_infos: List[FlowRunInfo], raise_on_line_failure: bool = False):
```

Comment:
I guess this is a batch specific logic, not a common logic in flow

Reply:
Yes. I think so. But this function is also used by `flow_executor` and runtime. I prefer to move it to a common utility instead of importing `BatchEngine` in `flow_executor`.

Reply:
Move it to execution_utils because flow_utils belongs to sdk